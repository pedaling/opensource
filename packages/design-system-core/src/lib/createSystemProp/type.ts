import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
