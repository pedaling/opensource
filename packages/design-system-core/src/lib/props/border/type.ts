import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
