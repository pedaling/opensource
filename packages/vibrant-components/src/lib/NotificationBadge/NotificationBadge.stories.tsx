import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { NotificationBadge } from './NotificationBadge';

export default {
  title: 'NotificationBadge',
  component: NotificationBadge,
  args: {
    kind: 'dot',
    size: 'md',
    borderColor: 'background',
  },
} as ComponentMeta<typeof NotificationBadge>;

export const Basic: ComponentStory<typeof NotificationBadge> = props => <NotificationBadge {...props} />;
