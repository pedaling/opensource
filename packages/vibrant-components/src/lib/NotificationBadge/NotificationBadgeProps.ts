import type { ResponsiveValue } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';
import type { BaseColorToken } from '@vibrant-ui/theme';

export type NotificationBadgeProps = {
  children?: never;
  borderColor?: BaseColorToken;
} & (
  | {
      kind: 'dot';
      size: ResponsiveValue<'lg' | 'md' | 'sm'>;
      count?: never;
    }
  | {
      kind: 'number';
      size?: never;
      count: number;
    }
);

export const withNotificationBadgeVariation = withVariation<NotificationBadgeProps>()(
  propVariant({
    props: [
      {
        name: 'count',
        keep: true,
      },
    ],
    variants: ({ count }) => {
      if (count && count > 9) {
        return {
          count: '9+',
          px: 4,
        };
      }

      return {
        width: 14,
      };
    },
  }),
  propVariant({
    props: [
      {
        name: 'size',
        responsive: true,
        default: 'sm' as const,
      },
    ],
    variants: {
      sm: {
        dotSize: 5,
      },
      md: {
        dotSize: 6,
      },
      lg: {
        dotSize: 8,
      },
    },
  }),
  propVariant({
    props: [
      {
        name: 'kind',
        keep: true,
      },
      {
        name: 'dotSize',
        responsive: true,
      },
    ],
    variants: ({ kind, dotSize }) => {
      if (kind === 'dot') {
        return {
          width: dotSize,
          height: dotSize,
          borderRadius: 4,
        };
      }

      return {
        width: 14,
        height: 14,
        borderRadius: 7,
        pb: 0.5,
      };
    },
  }),
  propVariant({
    props: [
      {
        name: 'borderColor',
      },
      {
        name: 'borderRadius',
        responsive: true,
        keep: true,
      },
    ],
    variants: ({ borderColor, borderRadius }) => {
      if (borderColor) {
        return {
          position: 'relative',
          pseudoBefore: {
            position: 'absolute',
            top: -2,
            left: -2,
            bottom: -2,
            right: -2,
            borderRadius: borderRadius + 2,
            borderWidth: 2,
            borderStyle: 'solid',
            borderColor,
          },
        } as const;
      }

      return {};
    },
  })
);
