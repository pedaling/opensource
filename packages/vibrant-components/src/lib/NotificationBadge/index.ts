export { NotificationBadge } from './NotificationBadge';
export type { NotificationBadgeProps } from './NotificationBadgeProps';
