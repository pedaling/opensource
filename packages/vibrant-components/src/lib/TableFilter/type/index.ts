export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
