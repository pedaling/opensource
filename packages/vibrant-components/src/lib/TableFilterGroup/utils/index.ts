import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
