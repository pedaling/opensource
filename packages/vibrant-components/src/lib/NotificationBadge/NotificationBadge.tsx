import { Box } from '@vibrant-ui/core';
import { Body } from '../Body';
import { withNotificationBadgeVariation } from './NotificationBadgeProps';

export const NotificationBadge = withNotificationBadgeVariation(({ kind, count, testId, ...props }) => (
  <Box
    display="inline-flex"
    backgroundColor="onViewError"
    whiteSpace="nowrap"
    alignItems="center"
    justifyContent="center"
    top={0}
    right={0}
    data-testid={testId}
    {...props}
  >
    {kind === 'number' && (
      <Body level={6} weight="medium">
        {count}
      </Body>
    )}
  </Box>
));
