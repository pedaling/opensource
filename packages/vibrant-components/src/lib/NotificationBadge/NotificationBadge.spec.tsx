import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { NotificationBadge } from './NotificationBadge';

describe('<NotificationBadge />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;
  let element: HTMLElement;

  describe('when NotificationBadge with number rendered', () => {
    beforeEach(async () => {
      renderer = render(<NotificationBadge data-testid="notificationBadge" kind="number" count={5} />);

      element = await renderer.getByTestId('notificationBadge');
    });

    it('count rendered', () => {
      expect(element.children[0].tagName).toBe('SPAN');

      expect(element.children[0].textContent).toBe('5');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when NotificationBadge with dot rendered', () => {
    beforeEach(async () => {
      renderer = render(<NotificationBadge data-testid="notificationBadge" kind="dot" size="md" />);

      element = await renderer.findByTestId('notificationBadge');
    });

    it('children is empty', () => {
      expect(element.childElementCount).toBe(0);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
