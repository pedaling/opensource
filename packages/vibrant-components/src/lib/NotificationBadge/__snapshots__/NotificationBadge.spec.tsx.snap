// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationBadge /> when NotificationBadge with dot rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #E10923;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  top: 0;
  right: 0;
  width: 6px;
  height: 6px;
  border-radius: 4px;
}

<div>
  <div
    class="emotion-0"
    data-testid="notificationBadge"
  />
</div>
`;

exports[`<NotificationBadge /> when NotificationBadge with number rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  background-color: #E10923;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  top: 0;
  right: 0;
  width: 14px;
  height: 14px;
  border-radius: 7px;
  padding-bottom: 0.5px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  font-family: Pretendard;
  text-align: left;
  color: #0C0C0C;
  font-size: 0.625rem;
  font-weight: 500;
  line-height: 0.75rem;
}

<div>
  <div
    class="emotion-0"
    data-testid="notificationBadge"
  >
    <span
      class="emotion-1"
    >
      5
    </span>
  </div>
</div>
`;
